docs/spl-docs/dedup-command.md

```diff
@@ -0,0 +1,177 @@
+
+Remove duplicate search results with the same `host` value.
+
+```
+... | dedup host
+```
+
+### 2. Remove duplicate results and sort results in ascending order
+
+Remove duplicate search results with the same `source` value and sort the results by the `timestamp` field in ascending order.
+
+```
+... | dedup source sortby +timestamp 
+```
+
+### 3. Remove duplicate results and sort results in descending order
+
+Remove duplicate search results with the same `source` value and sort the results by the `size` field in descending order.
+
+```
+... | dedup source sortby -size
+```
+
+### 4. Keep the first 3 duplicate results
+
+For search results that have the same `source` value, keep the first 3 that occur and remove all subsequent results.
+
+```
+... | dedup 3 source
+```
+
+### 5. Keep results that have the same combination of values in multiple fields
+
+For search results that have the same `source` AND `host` values, keep the first 2 that occur and remove all subsequent results.
+
+```
+... | dedup 2 source host
+```
+
+### 6. Remove only consecutive duplicate events
+
+Remove only consecutive duplicate events. Keep non-consecutive duplicate events. In this example duplicates must have the same combination of values the `source` and `host` fields.
+
+```
+... | dedup consecutive=true source host
+```
+
+### Use-Case Example
+
+**Identify and retain unique error logs for troubleshooting**
+
+**Problem:** A user wants to analyze error logs to identify unique errors based on the combination of error code and error message. They want to remove duplicate error logs to focus on unique errors for efficient troubleshooting.
+
+**Solution:** The `dedup` command can be used to remove duplicate events that have the same combination of error code and error message.
+
+**SPL command:**
+
+```
+index=error_logs 
+| dedup error_code error_message sortby -time
+| fields time, error_code, error_message, host, source
+```
+
+**Explanation:**
+1. The search starts by looking at the error logs with the `index=error_logs`.
+2. The `dedup` command removes duplicate events that have the same combination of `error_code` and `error_message`.
+3. The `sortby` command sorts the results by `time` in descending order to show the most recent unique errors first.
+4. The `fields` command selects the relevant fields (`time`, `error_code`, `error_message`, `host`, `source`) for output.
+
+**Example data:**
+
+```
+time                     error_code    error_message                host          source
+2024-07-12 09:15:23       404           Not Found                    server1       /var/log/httpd/error_log
+2024-07-12 09:16:00       500           Internal Server Error        server2       /var/log/httpd/error_log
+2024-07-12 09:17:45       404           Not Found                    server3       /var/log/httpd/error_log
+2024-07-12 09:18:30       503           Service Unavailable          server1       /var/log/httpd/error_log
+```
```

Comment:
Won't line 172 get deduped (from line 170)? Or if this is just the initial data, you should also show the results from the query.

Reply:
nit: the field names after `time` on line 169 are one space off